File: 0455-assign-cookies/0455-assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        if not s:
            return 0
        ans = 0
        s.sort(reverse=True)
        g.sort(reverse=True)
        print('s: ', s)
        print('g: ', g)
        index = 0
        for size in s:
            while size < g[index]:
                index += 1
                if index == len(g):
                    return ans
            ans += 1
            index += 1
            if index == len(g):
                return ans
        return ans

File: 0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        priority_queue<int> greedies;
        priority_queue<int> cookies;
        for (int greedy : g) {
            greedies.push(greedy);
        }
        for (int cookie : s) {
            cookies.push(cookie);
        }
        int ans = 0;
        while (!greedies.empty() && !cookies.empty()) {
            if (greedies.top() <= cookies.top()) {
                ans++;
                greedies.pop();
                cookies.pop();
            } else {
                greedies.pop();
            }
        }
        return ans;
    }
};